File: avance_en_python/paciente.py
# Importamos date para trabajar con fechas
from datetime import date

# Importamos las clases que Paciente necesita utilizar
from persona import Persona
from historia_clinica import HistoriaClinica


# Paciente hereda de la clase Persona
class Paciente(Persona):

    # Constructor de la clase Paciente
    def __init__(
        self,
        id,
        nombre,
        dni,
        telefono,
        fecha_nacimiento,
        direccion
    ):
        # Llamamos al constructor de la clase padre Persona
        # para inicializar id, nombre, dni y telefono
        super().__init__(id, nombre, dni, telefono)

        # Guardamos los atributos propios del paciente
        self.__fecha_nacimiento = fecha_nacimiento
        self.__direccion = direccion

        # Al crear un paciente también creamos
        # automáticamente su Historia Clínica
        self.__historia_clinica = HistoriaClinica(
            id,
            date.today(),
            "Sin antecedentes"
        )

    # Método para mostrar los datos del paciente
    def mostrar_datos_paciente(self):
        print("===== DATOS DEL PACIENTE =====")

        # obtener_datos() es un método heredado de Persona
        print(self.obtener_datos())

        # Mostramos los atributos propios de Paciente
        print(f"Fecha de nacimiento: {self.__fecha_nacimiento}")
        print(f"Dirección: {self.__direccion}")

    # Método para consultar la Historia Clínica del paciente
    def consultar_historial(self):

        # Llamamos al método consultar_historial()
        # del objeto HistoriaClinica
        self.__historia_clinica.consultar_historial()

    # Método que devuelve la Historia Clínica del paciente
    def get_historia_clinica(self):
        return self.__historia_clinica

File: src/policlinico/Paciente.java
package policlinico;

// Importamos LocalDate para trabajar con fechas
import java.time.LocalDate;

// La clase Paciente hereda de la clase Persona
public class Paciente extends Persona {

    // Atributos propios del paciente
    private LocalDate fechaNacimiento;
    private String direccion;

    // Cada paciente tiene una Historia Clínica
    private HistoriaClinica historiaClinica;

    // Constructor de la clase Paciente
    public Paciente(
            int id,
            String nombre,
            String dni,
            String telefono,
            LocalDate fechaNacimiento,
            String direccion) {

        // Llamamos al constructor de la clase padre Persona
        // para inicializar id, nombre, dni y telefono
        super(id, nombre, dni, telefono);

        // Inicializamos los atributos propios de Paciente
        this.fechaNacimiento = fechaNacimiento;
        this.direccion = direccion;

        // Al crear un paciente, también se crea su Historia Clínica
        // Usamos el mismo id del paciente,
        // la fecha actual y un valor inicial para antecedentes
        historiaClinica = new HistoriaClinica(
                id,
                LocalDate.now(),
                "Sin antecedentes"
        );
    }

    // Método para mostrar los datos principales del paciente
    public void mostrarDatosPaciente() {

        System.out.println("===== DATOS DEL PACIENTE =====");

        // Este método viene heredado de la clase Persona
        System.out.println(obtenerDatos());

        // Mostramos los datos propios de Paciente
        System.out.println("Fecha de nacimiento: " + fechaNacimiento);
        System.out.println("Dirección: " + direccion);
    }

    // Método para consultar la Historia Clínica del paciente
    public void consultarHistorial() {

        // Llamamos al método consultarHistorial()
        // del objeto historiaClinica
        historiaClinica.consultarHistorial();
    }

    // Método que devuelve la Historia Clínica del paciente
    // Esto permite trabajar con ella desde otras partes del programa
    public HistoriaClinica getHistoriaClinica() {
        return historiaClinica;
    }
}

    public HistoriaClinica getHistoriaClinica() {
        return historiaClinica;
    }
}
